app/main.py

```diff
@@ -1,3 +1,71 @@
+from __future__ import annotations
+
+
 class Distance:
-    # Write your code here
-    pass
+    def __init__(self, km: float | int) -> None:
+        self.km = km
+
+    def __str__(self) -> str:
+        return f"Distance: {self.km} kilometers."
+
+    def __repr__(self) -> str:
+        return f"Distance(km={self.km})"
+
+    def __add__(self, other: Distance | float | int) -> Distance:
+        other = Distance.to_distance(other)
+        return Distance(self.km + other.km)
+
+    def __iadd__(self, other: Distance | float | int) -> Distance:
+        other = Distance.to_distance(other)
+        self.km += other.km
+        return self
+
+    def __mul__(self, other: float | int) -> Distance:
+        km = self.to_km(other)
+        return Distance(self.km * km)
+
+    @staticmethod
+    def to_km(dist: int | float) -> float | int:
+        if not (
+                isinstance(dist, int)
+                or isinstance(dist, float)
+        ):
+            raise TypeError
+        return dist
```

Comment:
The `to_km` method should return a float, not an int. The method is meant to ensure the input is a numeric type, but it should not change the type to int if it is already a float.